webpack/worker/worker_adapter.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * @fileoverview wrapper program around the Webpack CLI.
+ *
+ * It intercepts the Bazel Persistent Worker protocol, using it to
+ * remote-control webpack cli. It tells the compiler process to
+ * consolidate file changes only when it receives a request from the worker
+ * protocol.
+ *
+ * See https://medium.com/@mmorearty/how-to-create-a-persistent-worker-for-bazel-7738bba2cabb
+ * for more background on the worker protocol.
+ */
+const worker = require("@bazel/worker");
+const cp = require('child_process');
+const fs = require('fs');
+
+const webpackCli = require("webpack-cli");
+
+/** @typedef {{type: "built" | "error" | "ready"}} IPCMessage */
+
+function main() {
+  const MNEMONIC = 'webpack';
+  if (worker.runAsWorker(process.argv)) {
+    const webpackCliPath = require.resolve("webpack-cli/bin/cli.js");
```

Comment:
Consider splitting both the if and else into seperate functions for readability

Reply:
That’s reasonable. thanks ☺️